WIP UPSTREAM <carry>: Rebase code to kfp 2.2.0

rimolive opened the pull request:
**Description of your changes:**
Resolves #[RHOAIENG-6670](https://issues.redhat.com//browse/RHOAIENG-6670)

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 
<!--
   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.
-->

gregsheremeta commented on manifests/opendatahub/README.md:
```diff
@@ -9,19 +9,31 @@ Data Science Pipelines is the Open Data Hub's pipeline solution for data scienti
 2. OpenShift Pipelines 1.7.2 or higher needs to be installed on the cluster
-3. The Open Data Hub operator needs to be installed
-4. The default installation namespace for Data Science Pipelines is `odh-applications`. This namespace will need to be created. In case you wish to install in a custom location, create it and update the kfdef as documented below.
+3. The [oc client](https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.11.3/openshift-client-linux.tar.gz) is installed and you are logged into this cluster.
```
nit: link to `latest` instead of a specific version
```suggestion
3. The [oc client](https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/latest/openshift-client-linux.tar.gz) is installed and you are logged into this cluster.
```

accorvin commented on manifests/opendatahub/README.md:
```diff
@@ -9,19 +9,31 @@ Data Science Pipelines is the Open Data Hub's pipeline solution for data scienti
 
 1. The cluster needs to be OpenShift 4.9 or higher
 2. OpenShift Pipelines 1.7.2 or higher needs to be installed on the cluster
```
IMO we should make it obvious how to install OpenShift Pipeilnes (similarly to how we do for the Open Data Hub below). Perhaps we could simply add this sub-bullet:
```
You can do this by installing the OpenShift Pipelines operator from Operator Hub using the same process as followed below for the Open Data Hub operator.
```

accorvin replied:
That seems like a reasonable thing to do.

One can certainly argue that this is out of scope for things our team should be documenting, but making this document as idiot proof as possible will make it much easier for future team members/idiots.

accorvin replied:
Thanks!

gregsheremeta commented on manifests/opendatahub/README.md:
````diff
@@ -9,19 +9,31 @@ Data Science Pipelines is the Open Data Hub's pipeline solution for data scienti
     ```
     The value of `<kfdef_namespace>` should match the namespace field of the kfdef that you applied.
-4. Alternatively, you can access the route via the console. To do so:
+    You may need to append the protocol to the route (ie `https://`), but this can otherwise be pasted into a browser to access the UI
````
nice touch. Chrome will usually think I mean http when I do this and forget.

accorvin commented:
/approve

HumairAK commented:
/lgtm

gregsheremeta commented:
/lgtm

DharmitD reviewed (commented):
/lgtm

TheMichaelHu committed:
feat(components): Add better docstrings for AutoSxS

PiperOrigin-RevId: 591051675

Commit:
test(components): INTERNAL

PiperOrigin-RevId: 591075392

connor-mccarthy committed:
feat(sdk): add subprocess task handler #localexecution (#10302)

* add subprocess task handler

* response to review feedback

Commit:
fix(components): Use `large_model_reference` as `model_reference_name` when uploading models from `preview.llm.rlhf_pipeline` instead of hardcoding value as `text-bison@001`

PiperOrigin-RevId: 591346782

connor-mccarthy committed:
test(sdk): add `kfp-kubernetes` execution tests (#10304)

Commit:
fix(components): Resolve unique model display name on each `preview.llm.rlhf_pipeline` run instead of reusing cached result

PiperOrigin-RevId: 591365087

Commit:
fix(components): Upload the tuned adapter to Model Registry instead of model checkpoint from `preview.llm.rlhf_pipeline`

PiperOrigin-RevId: 591393116

connor-mccarthy committed:
feat(sdk): add local execution output collection #localexecution (#10325)

* feat(sdk): add local execution output collection #localexecution

* update ExecutorInput processing

* address review feedback

* fix exception handling test

connor-mccarthy committed:
chore(sdk): fix use of invalid escape sequence in tests (#10310)

connor-mccarthy committed:
feat(sdk): add local execution logging #localexecution (#10326)

Commit:
chore(components): Fix argument description

PiperOrigin-RevId: 592033936

TobiasGoerke committed:
feat(backend): preserve querystring in pipeline root (fixes #10318) (#10319)

* feat: preserve querystring in pipeline root

* refactor: create AppendToPipelineRoot
Also apply to client.go

* feat: remove query string from URIs (#1)

* feat: remove query string from URIs

* refactor(GenerateOutputURI): move and preserve comments

Commit:
feat(components): change output format to allow possible post eval

PiperOrigin-RevId: 592257929

zijianjoy committed:
fix(frontend): Add disableParsingRawHTML option for markdown-to-jsx component (#10315)

connor-mccarthy committed:
feat(sdk): add DockerRunner #localexecution (#10328)

connor-mccarthy committed:
feat(sdk): add special `dsl.OutputPath` read logic #localexecution (#10334)

connor-mccarthy committed:
chore(sdk): write local execution logs to stdout #localexecution (#10330)

connor-mccarthy committed:
feat(sdk): support local Container Component execution #localexecution (#10333)

* support Container Components

* address review feedback

TheMichaelHu committed:
No public description

PiperOrigin-RevId: 592702825

Commit:
feat(components): Output errors as a separate table from Arbiter

PiperOrigin-RevId: 592769441

TheMichaelHu committed:
docs(components): Fix AutoSxS docstring formatting

PiperOrigin-RevId: 594058915

connor-mccarthy committed:
chore(components): add `json_escape` placeholder util

PiperOrigin-RevId: 595174285

connor-mccarthy committed:
fix(sdk): fix presentation of strings in local execution #localexecution (#10353)

connor-mccarthy committed:
fix(sdk): remove redundant newline character in local `DockerRunner` logs (#10354)

connor-mccarthy committed:
test(sdk): improve KFP SDK local runner test safety #localexecution (#10336)

DharmitD committed:
feat(backend): Enable logging for KFP components (#10288)

connor-mccarthy committed:
feat(sdk): remove local execution feature flag #localexecution (#10355)

connor-mccarthy committed:
fix(sdk): permit empty local execution outputs #localexecution (#10338)

connor-mccarthy committed:
feat(sdk): support Concat and IfPresent placeholder in local container component execution #localexecution (#10348)

* feat(sdk): support Concat and IfPresent placeholder in local container component execution #localexecution

* address review feedback

* fix test

connor-mccarthy committed:
test(sdk): add test for local execution of `None` default parameter #localexecution (#10339)

connor-mccarthy committed:
chore(sdk): release KFP SDK 2.5.0 (#10364)

connor-mccarthy committed:
chore(sdk): depend on protobuf 4 in `kfp-pipeline-spec` (#10305)

* chore(sdk): depend on protobuf 4 in `kfp-pipeline-spec`

* bump kfp-pipeline-spec version

TheMichaelHu committed:
docs(components): Document AutoML Tables util functions

PiperOrigin-RevId: 596735195

Commit:
feat(components): Implement the feature store grounding pipeline

PiperOrigin-RevId: 597032159

jsondai committed:
feat(components): update eval pipeline documentation to clarify the required pipeline parameters

PiperOrigin-RevId: 597080455

connor-mccarthy committed:
docs(components): update BigqueryQueryJobOp docs on container args limit

PiperOrigin-RevId: 597579180

connor-mccarthy committed:
chore(components): GCPC 2.8.1 release

PiperOrigin-RevId: 597580166

Tomcli committed:
feat(kubernetes_platform): Update kubernetes_platform go package to include pod labels and annotations (#10357)

e pod metadata

connor-mccarthy committed:
chore(sdk): depend on protobuf 4 in `kfp` and `kfp-kubernetes` (#10307)

Commit:
feat(components): Implement new output format of inference component

PiperOrigin-RevId: 597621035

connor-mccarthy committed:
chore(components): rollback GCPC 2.8.1 release

PiperOrigin-RevId: 597636452

connor-mccarthy committed:
chore(sdk): release KFP SDK 2.6.0 (#10386)

Commit:
chore(components): Add v1.model.ModelGetOp components

PiperOrigin-RevId: 597653399

connor-mccarthy committed:
chore(sdk): release kfp-kubernetes 1.1.0 (#10387)

connor-mccarthy committed:
chore(sdk): update `kfp-kubernetes` docs versions and release scripts (#10388)

connor-mccarthy committed:
docs(sdk): fix `kfp-kubernetes` docs build error (#10389)

TheMichaelHu committed:
chore(components): Sync AutoML components

PiperOrigin-RevId: 597743182

jsondai committed:
fix(components): Update base image for KFP lightweight component for VPC SC compliance

PiperOrigin-RevId: 598888543

Commit:
Fix metrics visualization v2 sample (#10399)

connor-mccarthy committed:
docs(sdk): add kfp.local to reference docs (#10395)

connor-mccarthy committed:
chore(components): migrate GCPC to protobuf 4; require KFP>=2.6.0

PiperOrigin-RevId: 599639627

Commit:
feat(components): Support scheduling and labels in utils.build_payload

PiperOrigin-RevId: 599904659

Commit:
chore(components): Sync AutoML components

PiperOrigin-RevId: 600091779

connor-mccarthy committed:
feat(components): support aliases arg in ModelUploadOp

PiperOrigin-RevId: 600096017

connor-mccarthy committed:
fix string quotes (#10413)

Commit:
docs(components): Update AutoSxS pipeline to use "question_answering" as task name instead of "question_answer", where "question_answer" is still supported, but deprecated
chore(components): Update RLHF and AutoSxS image tags

PiperOrigin-RevId: 600532640

Commit:
fix(components): Update base image for KFP lightweight component for VPC SC compliance

PiperOrigin-RevId: 600622671

connor-mccarthy committed:
chore(sdk): local task execution refactor + cleanup (#10420)

hellojunha committed:
fix(backend): MLMD pagination on getting executions of DAG (#10396)

Commit:
fix(components): Only run `preview.llm.bulk_inference` after tuning third-party models with RLHF

PiperOrigin-RevId: 601226133

jsondai committed:
feat(components): Update LLM Evaluation Pipelines to use `text-bison@002` model by default

PiperOrigin-RevId: 601509035

connor-mccarthy committed:
chore(components): GCPC 2.9.0 Release

PiperOrigin-RevId: 601525956

connor-mccarthy committed:
chore(components): fix GCPC documentation dependencies issue

PiperOrigin-RevId: 601574718

TheMichaelHu committed:
feat(components): Use a single inference component for AutoSxS

PiperOrigin-RevId: 601877680

connor-mccarthy committed:
feat(sdk): support local pipeline execution (#10423)

* local pipeline implementation

* address review feedback

KCFindstr committed:
fix(components): Write model resource_name to the output of training pipeline remote runner

PiperOrigin-RevId: 602426716

connor-mccarthy committed:
feat(sdk): support f-strings in local pipeline execution (#10435)

connor-mccarthy committed:
support dsl.importer locally; resolve merge conflicts (#10431)

Commit:
fix(components): Update text generation pipeline input description

PiperOrigin-RevId: 602860006

connor-mccarthy committed:
feat(sdk): support local execution of pipelines in pipelines (#10440)

HumairAK committed:
fix(backend): correct run field map col names (#10430)

* fix(backend): correct field map col names

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* chore(backend): add tests for api field map

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

---------

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

TheMichaelHu committed:
chore(components): Create module containing AutoSxS and RLHF image tag

PiperOrigin-RevId: 603765313

Commit:
No public description

PiperOrigin-RevId: 604419526

TheMichaelHu committed:
chore(components): Update component naming in AutoSxS implementation

PiperOrigin-RevId: 604678269

KevinGrantLee committed:
feat(sdk): Support dsl.ParallelFor over list of Artifacts (#10441)

* samples message

* update pr number

* Split LoopArgument into LoopParameterArgument and LoopArtifactArgument

* formatting

* address some comments

* resolve release notes conflict

* flatten loops in pipeline_spec_builder

* update artifact type checking logic

* simplify artifact checking logic

* re-add issubtype_of_artifact

* move name_is_loop_argument to for_loop_test.py

* simplify LoopArtifactArgument is_artifact_list=False logic

* update typeerror

* typo

* typo

* small fix

* formatting

* formatting

* remove issubtype_of_artifact()

* small changes

* assert LoopArtifactArgument channel.is_artifact_list is True

* remove whitespace

* remove newline

* Update single artifact check and error message

* formatting

* add unit test for is_artifact_list==False

* formatting

* update valueerror test.

* typo

* regex formatting

* formatting

JosepSampe committed:
feat(kubernetes_platform): Update kubernetes_platform go package to include ImagePullSecrets (#10410)

revit13 committed:
feat(kubernetes_platform): Update kubernetes_platform go package to include imagePullPolicy. (#10416)

Signed-off-by: Revital Sur <eres@il.ibm.com>

Commit:
No public description

PiperOrigin-RevId: 605012378

rickyxie0929 committed:
fix(sdk): fixes type issues for ParallelFor. Fixes #9366 (#10436)

* fix type issues

* fix type issue

* fix format

* fix failed test

* fix format

* delete comments

* resolve comments

* resolve comments

* resolve format

* resolve import

* move unnecessary file

* resolve compiler_test failures

* resolve comments

* remove unnecessary imports

* fix format sort

* fix nits

* add new compiled yaml file

* solve merge conflicts

* solve conflicts

* format

* sort

* resolve comments

* resolve comments

Commit:
feat(components): Add RLAIF pipeline to preview

PiperOrigin-RevId: 605396378

connor-mccarthy committed:
fix(sdk): fix bug where `dsl.OneOf` with multiple consumers cannot be compiled (#10452)

TheMichaelHu committed:
chore(components): Use new module for looking up ReFINED and AutoSxS image tags

PiperOrigin-RevId: 605457014

JosepSampe committed:
feat(Backend + SDK): Update kfp backend and kubernetes sdk to support ImagePullSecrets (#10427)

* Update kfp backend and kubernetes sdk to support ImagePullSecrets

* update go.mod go.sum and csv files

* update image_pull_secret method name

* update unit tests

* update apiserver.csv file

* update set_image_pull_secrets name

abaland committed:
Fix(backend): fix timeout for internal server error. Fixes #10267 (#10439)

* Fix(backend): fix timeout for internal server error

* Fix(backend): fix timeout for internal server error (variable rename)

cybernagle committed:
fix(backend): get pipeline by name is broken due to version typo, Fixes #9940 (#10268)

function getPipelineByName have a version typo:
V1beta1 -> v1beta1

roytman committed:
feat(kubernetes_platform): Update kubernetes_platform go package to include ConfigMaps as volumes and as env variables. (#10400)

* Update kubernetes_platform go package to include ConfigMaps as volumes and as env variables

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* add ConfigMapKeyToEnvMap

* change config_map_key string

* add list of map

* rename "CmAsVolume" to "ConfigMapAsVolume" and "CmAsEnv" to "ConfigMapAsEnv"

---------

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

Tomcli committed:
feat(Backend + SDK): Update kfp backend and kubernetes sdk to support pod labels and annotations (#10393)

* update kfp kubernetes sdk to include pod labels and annotations

* fix unit test output order

* add podmetadata changes

* update argo compiler to support pod metadata

* update tests

* update go mod to use the latest kubernetes_platform package

* update licenses

* address comments

* update kubernetes_platform package to include the latest spec

---------

Co-authored-by: Chen Sun <chensun@users.noreply.github.com>

Commit:
chore(components): Update AutoSxS and RLHF image tags

PiperOrigin-RevId: 606330905

Tomcli committed:
feat(kubernetes_platform): Add ActiveDeadlineSeconds(timeout) to the kubernetes platform spec (#10464)

connor-mccarthy committed:
chore(components): support jinja 3 in GCPC

PiperOrigin-RevId: 606703078

ananth102 committed:
test(components): Reduce sagemaker component test flakiness (#10225)

Signed-off-by: ananth102 <abashyam@amazon.com>

connor-mccarthy committed:
chore(sdk): release KFP SDK 2.7.0 (#10461)

TheMichaelHu committed:
fix(components): Use PipelineJob location in AutoSxS components, add init file

PiperOrigin-RevId: 607055407

Tomcli committed:
feat(kubernetes_platform): Add k8s FieldPath as env to the kubernetes_platform (#10485)

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

chensun committed:
chore: Add Tomcli as a backend approver (#10490)

Signed-off-by: Chen Sun <chensun@users.noreply.github.com>

Commit:
chore(components): Sync AutoML components

PiperOrigin-RevId: 607435076

Tomcli committed:
chore(README): Update Kubeflow Pipelines on Tekton blog (#10482)

Commit:
chore(components): Rename several `_implementation.llm` components

PiperOrigin-RevId: 607487816

droctothorpe committed:
feat(kubernetes_platform): Update kubernetes_platform go package to i… (#10442)

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: edmondop <edmondo.porcu@gmail.com>
Co-authored-by: tarat44 <32471142+tarat44@users.noreply.github.com>

connor-mccarthy committed:
docs(components): internal

PiperOrigin-RevId: 607802006

connor-mccarthy committed:
chore(components): bump highest support KFP SDK version in GCPC to KFP SDK 2.7.0

PiperOrigin-RevId: 608646229

hbelmiro committed:
fix(backend): fixes "cannot save parameter" error message. Fixes #9678 (#10459)

Signed-off-by: hbelmiro <helber.belmiro@gmail.com>

Commit:
fix(rlhf): Supporting adapter only output for reward model training

PiperOrigin-RevId: 608740017

Commit:
chore(components): internal

PiperOrigin-RevId: 609081795

Commit:
docs(components): internal

PiperOrigin-RevId: 609107204

Commit:
feat(components): Added experimental args to batch_prediction_pairwise component

PiperOrigin-RevId: 609129336

Commit:
chore(components): release GCPC SDK 2.10.0

PiperOrigin-RevId: 609459972

droctothorpe committed:
feat(Backend + SDK): Update kfp backend and kubernetes sdk to support tolerations (#10471)

* feat(Backend + SDK): Update kfp backend and kubernetes sdk to support tolerations

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: edmondop <edmondo.porcu@gmail.com>
Co-authored-by: tarat44 <32471142+tarat44@users.noreply.github.com>

* Address PR review 1

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: edmondop <edmondo.porcu@gmail.com>
Co-authored-by: tarat44 <32471142+tarat44@users.noreply.github.com>

* Refactor add_toleration to use Python primitives

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: edmondop <edmondo.porcu@gmail.com>
Co-authored-by: tarat44 <32471142+tarat44@users.noreply.github.com>

* Update go.mod to pull in latest kubernetes_platform package

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: edmondop <edmondo.porcu@gmail.com>
Co-authored-by: tarat44 <32471142+tarat44@users.noreply.github.com>

---------

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: edmondop <edmondo.porcu@gmail.com>
Co-authored-by: tarat44 <32471142+tarat44@users.noreply.github.com>

Commit:
feat(components): Add configurable image prefix to llm utility method

PiperOrigin-RevId: 609560776

Commit:
chore(components): internal change

PiperOrigin-RevId: 609757226

Commit:
feat(components): Add CMEK support to `preview.llm.rlhf_pipeline`

PiperOrigin-RevId: 609832020

roytman committed:
feat(Backend + SDK): Update kfp backend and kubernetes sdk to support ConfigMaps as volumes and as env variables (#10483)

* Update kfp backend and kubernetes sdk to support ConfigMaps as volumes and as env

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* update go.mod, apiserver.csv and driver.csv

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* add test/snapshot/data files

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* fix tests

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* go mod tidy

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* update backend/third_party_licenses/apiserver.csv

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* update backend/third_party_licenses/apiserver.csv

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* fix comments

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* fix comments

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* update go.mod, apiserver.csv and driver.csv

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

---------

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

Commit:
feat(components): Release Forecasting training pipelines to V1 namespace

PiperOrigin-RevId: 610830518

KevinGrantLee committed:
fix(sdk): Prevents dsl.ParallelFor over single parameter from compiling. (#10494)

* fix(sdk): Prevents dsl.ParallelFor over single paramter from compiling.

* fix(sdk): Prevents dsl.ParallelFor over single paramter from compiling.

* update PR number in release notes

* formatting

* Add compiler_test.py test for single param compile failure

* Update some docstrings and add todo

* formatting

* Update sdk/python/kfp/compiler/compiler_test.py

Co-authored-by: Connor McCarthy <mccarthy.connor.james@gmail.com>

* Update sdk/python/kfp/compiler/compiler_test.py

Co-authored-by: Connor McCarthy <mccarthy.connor.james@gmail.com>

* Update sdk/python/kfp/dsl/for_loop.py

Co-authored-by: Connor McCarthy <mccarthy.connor.james@gmail.com>

* Use print_and_return and other small changes

* typo

* typo

---------

Co-authored-by: Connor McCarthy <mccarthy.connor.james@gmail.com>

Commit:
fix(components): Pass tuned model checkpoint to inference pipeline after RLHF tuning

PiperOrigin-RevId: 610918020

KevinGrantLee committed:
test: Add ParallelFor compile test over single artifact. (#10531)

* .

* formatting

Commit:
fix(components): rename custom task calibration_score_rubric -> score_rubric

PiperOrigin-RevId: 611161020

jsondai committed:
chore(components): Switch default machine type for LLM Text Generation Eval pipeline and components to e2-standard-4

PiperOrigin-RevId: 612531671

TheMichaelHu committed:
fix(components): Propagate location to sub-components in AutoSxS

PiperOrigin-RevId: 612553652

Tomcli committed:
feat(backend + SDK): Add backend and SDK support to use Kubernetes FieldPath as env (#10496)

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

Commit:
chore(components): Change docker image URI used by `preview.llm` pipelines

PiperOrigin-RevId: 612662160

hbelmiro committed:
fix(docs): Updated legal info due to migration from CLA to DCO (#10501)

* Updated legal info due to migration from CLA to DCO

Signed-off-by: hbelmiro <helber.belmiro@gmail.com>

* Updated DCO link

Signed-off-by: hbelmiro <helber.belmiro@gmail.com>

---------

Signed-off-by: hbelmiro <helber.belmiro@gmail.com>

Tomcli committed:
feat(backend + SDK): Add Backend and SDK support for timeout in pod spec (#10481)

* Add backend and sdk support for pod spec timeout

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

* fix conflicts

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

---------

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

Commit:
fix(components): Return None as sliced feature attribution values for the classes which are not predicted in bp outputs

PiperOrigin-RevId: 612920651

Commit:
feat(components): Add CMEK validation to `preview.llm.infer_pipeline`

PiperOrigin-RevId: 612956960

revit13 committed:
feat(Backend + SDK): Update kfp backend and kubernetes sdk to support ImagePullPolicy (#10417)

* feat(Backend + SDK): Update kfp backend and kubernetes sdk to support ImagePullPolicy.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Fix format.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Update apiserver.csv.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Update licenses.

Signed-off-by: Revital Sur <eres@il.ibm.com>

---------

Signed-off-by: Revital Sur <eres@il.ibm.com>

connor-mccarthy committed:
docs(components): fix `create_custom_training_job_from_component` docs rendering

PiperOrigin-RevId: 612985431

Commit:
feat(components): Implement the train time evaluation in reward model training. With the train time eval dataset available, the pipeline outputs the accuracy and cross entropy metrics to the log

PiperOrigin-RevId: 613057150

PratyushaRavi committed:
fix(samples): Updated samples/core to V2 (#9879)

* Updated output_a_directory.py to V2

* Update output_a_directory_test.py to V2

* Update parallel_join.py to V2

* Update multiple_outputs.ipynb to V2

* Update multiple_outputs_test.py to V2

* Updated kfp_env_validation to V2

* Updated loop_parallelism to V2

Commit:
chore(components): Consolidate validation components in `preview.llm.rlhf_pipeline`

PiperOrigin-RevId: 613366137

Tomcli committed:
feat(backend): Upgrade go version to 1.20 (#10502)

* upgrade go version to 1.21

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

* upgrade integration test to go 1.21

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

* refresh go mod tidy

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

* fix license

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

* update go-sqlite3 to v1.14.19 to support go 1.21 on cache server

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

* downgrade go version to 1.20 and revert dockerfile upgrade

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

---------

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

Commit:
docs(components): Modify the GetModel documentation

PiperOrigin-RevId: 614771557

Commit:
chore(components): Update AutoSxS and RLHF image tags

PiperOrigin-RevId: 614785091

champon1020 committed:
fix(backend): Fixes response status of http error code when uploading duplicate pipeline [Fixes  #10311] (#10546)

Validate the error code of pipeline creation in order to return
the status conflict when the error represents AlreadyExists.

Signed-off-by: champon1020 <nagatelu1020@gmail.com>

Commit:
chore: Change stalebot rules (#10547)

daysUntilClose is set to 90 days. This is too much for an issue to keep opened as there is daysUntilStale set to 90 days.
That would keep an issue opened for 6 months, and now there are +600 issues opened in the repository.

We need to start working on keeping the repository healthy.

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

revit13 committed:
fix(kubernetes_platform): Add optional field to SecretAsVolume and ConfigMapAsVolume. Fixes #10548 (#10549)

* fix(kubernetes_platform): Add optional field to SecretAsVolume and ConfigMapAsVolume.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Update comment.

Signed-off-by: Revital Sur <eres@il.ibm.com>

---------

Signed-off-by: Revital Sur <eres@il.ibm.com>

TheMichaelHu committed:
feat(components): Add CMEK support to AutoSxS pipeline

PiperOrigin-RevId: 615498240

Commit:
feat(components): Update _LLM_EVAL_VERSION to v0.6

PiperOrigin-RevId: 615562899

jsondai committed:
chore(components): update container image of endpoint batch predict component for vulnerability patch

PiperOrigin-RevId: 615593420

chensun committed:
chore(backend): Update kfp driver and launcher images (#10561)

Signed-off-by: Chen Sun <chensun@users.noreply.github.com>

Commit:
fix(components): Add relevant component and pipeline inputs/outputs to support creating ModelEvaluations as part of the AutoSxS Metrics component

PiperOrigin-RevId: 615675169

Commit:
feat(components): Release Forecasting training pipelines to V1 namespace

PiperOrigin-RevId: 615914679

chensun committed:
chore(release): bumped version to 2.1.0

Commit:
feat(components): Add location validation to `preview.llm.rlhf_pipeline`

PiperOrigin-RevId: 616229944

chensun committed:
Revert "chore(release): bumped version to 2.1.0"

This reverts commit 4d90770dd319b7b342d601a3f04562f46301d583.

Commit:
chore(components): Testing fix

PiperOrigin-RevId: 616240307

junggil committed:
fix(samples): Update resource_spec, retry, secret samples to v2 pipelines (#9876)

* Update resource_spec, retry, secret samples to v2 pipelines

* Update resource_spec, retry, secret samples to v2 pipelines

amadhusu committed:
fix: Modified the comment/text for pipeline_version_id (#10581)

Signed-off-by: Achyut Madhusudan <amadhusu@redhat.com>

Commit:
chore(components): Update AutoSxS and RLHF image tags

PiperOrigin-RevId: 616895855

chensun committed:
chore: Update api-generator to use golang 1.20 (#10580)

Signed-off-by: Chen Sun <chensun@users.noreply.github.com>

connor-mccarthy committed:
fix(docs): make full version dropdown show on all KFP SDK docs versions (#10577)

Signed-off-by: connor-mccarthy <mccarthy.connor.james@gmail.com>

Commit:
feat(components): Implement new component to preprocess and validate inputs for rlhf

PiperOrigin-RevId: 617285265

Commit:
chore(components): GCPC 2.11.0 Release

PiperOrigin-RevId: 617287233

Commit:
fix(components): Fix missing pipeline parameters

PiperOrigin-RevId: 617579452

chensun committed:
chore: Update release image (#10599)

Signed-off-by: Chen Sun <chensun@users.noreply.github.com>

Commit:
fix(components): Remove the unused resolve_candidate_columns from function_based

PiperOrigin-RevId: 617802429

amadhusu committed:
fix: Modified the swagger json files according to the modified proto files. (#10591)

Signed-off-by: Achyut Madhusudan <amadhusu@redhat.com>

Commit:
feat(components): Copy text generation eval and text classification evak pipelines from preview to v1

PiperOrigin-RevId: 618017914

Commit:
docs(components): internal

PiperOrigin-RevId: 618313694

chensun committed:
chore(release): bumped version to 2.1.0

Commit:
feat(components): Report TensorBoard metrics for `preview.llm.rlhf_pipeline` in real time

PiperOrigin-RevId: 618924675

abaland committed:
feat(kubernetes_platform): Update kubernetes_platform go package to include generic ephemerl volume (#10602)

* feat(kubernetes_platform): Update kubernetes_platform go package to include Generic Ephemeral Volume resources
Signed-off-by: abaland <abaland@indeed.com>

* feat(kubernetes_platform): Update kubernetes_platform go package to include Generic Ephemeral Volume resources (include metadata)
Signed-off-by: abaland <abaland@indeed.com>

freefood89 committed:
fix(metadata envoy): upgrade envoy and config from 1.12 to 1.27 (#10589)

Signed-off-by: Rentaro Matsukata <freefood89@users.noreply.github.com>

Commit:
fix(components): Remove the unused resolve_data_paths from function_based

PiperOrigin-RevId: 619026115

Commit:
chore: Add Question issue template (#10557)

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

Commit:
feat(kubernetes_platform): Update kubernetes_platform go package to include node affinities and pod (anti)affinities (#10583)

* feat(kubernetes_platform): Update kubernetes_platform go package to include nodeaffinities and pod (anti)affinities

Signed-off-by: Cornelis Boon <cjidboon94@gmail.com>

* rename affinity objects and fields to match k8s spec semantics

Signed-off-by: Cornelis Boon <cjidboon94@gmail.com>

* rename *AffinityRule -> *AffinityTerm and add missing affinity data

Signed-off-by: Cornelis Boon <cjidboon94@gmail.com>

---------

Signed-off-by: Cornelis Boon <cjidboon94@gmail.com>

Commit:
chore(components): Add test machine spec support to `preview.llm` pipelines

PiperOrigin-RevId: 619378459

Commit:
fix(components): Ensure `preview.llm.rlhf_pipeline` runs if no `tensorboard_id` is provided

PiperOrigin-RevId: 619646459

Commit:
fix(components): Remove the unused functions from function_based

PiperOrigin-RevId: 619665186

Commit:
feat(components): Add model name preprocess component; Use publisher model if user uploaded model is non-tuned

PiperOrigin-RevId: 619714978

Commit:
feat(components): add task_type as a parameter to rlaif

PiperOrigin-RevId: 619744783

Commit:
feat(components): AutoSxS GA pending release

PiperOrigin-RevId: 619748191

Commit:
fix(components): Update service account comment

PiperOrigin-RevId: 619982662

connor-mccarthy committed:
chore(components): GCPC 2.12.0 Release

PiperOrigin-RevId: 619990671

Tomcli committed:
Update loop_output.py example for the new parallel loop type requirement (#10637)

Signed-off-by: tomcli <tommy.chaoping.li@ibm.com>

Commit:
feat(components): Added support for text-bison@002 to preview.llm.rlhf_pipeline

PiperOrigin-RevId: 620516628

connor-mccarthy committed:
No public description

PiperOrigin-RevId: 620699930

TheMichaelHu committed:
fix(components): Make AutoSxS autorater_prompt_parameters required

PiperOrigin-RevId: 620855438

Commit:
docs(components): internal

PiperOrigin-RevId: 621215497

Tomcli committed:
chore(components): Update kserve component to v0.12.0 (#10652)

Signed-off-by: tomcli <tommy.chaoping.li@ibm.com>

gmfrasca committed:
fix(backend): Update backend common code and integration tests with updated API Service Params (#10640)

* fix(backend): Refactor backend common code to use updated API Service Params

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* fix(backend): Fix Backend int tests to use updated API Service Params

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* WIP: fix(backend): Manually correct the proto schemes to include https

- **NOTE**: this was manually updated, tested, and verified locally
            uploading for CI check
- It appears when regenerating the backend API, something in the
  generation libraries changed and now default the scheme to just
  http, not http+https, which appears to break tests.
- Need to figure out what options to provide api generators to revert
  DefaultScheme to include https again automatically

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* chore: Update small syntax change in SWF expected test result check

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* chore: Update backend .proto files to include http and https scheme

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

---------

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

revit13 committed:
fix(Backend + SDK): Add missing optional field to SecretAsVolume and … (#10550)

* fix(Backend + SDK): Add missing optional field to SecretAsVolume and ConfigMapAsVolume.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Update after rebase.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Update after rebase.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Update after merge.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Updates after merge with master branch.

Signed-off-by: Revital Sur <eres@il.ibm.com>

---------

Signed-off-by: Revital Sur <eres@il.ibm.com>

rawc0der committed:
chore(manifests): refactor manifests for kustomize5 compatibility. Part of #10053 (#10087)

* refactor ml-pipeline component for kustomize5 compatibility

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* rm bk folder

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix kustomizations using auto kustomize edit

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix env/aws

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix kustomize version v5.0.3 in tests

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* minor changes to manifest apply method in pipeline deployment script

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix kustomize release

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix archive

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix bin

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* replace patchStrategicMerge refs in test manifests

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* replace kustomize version with 5.2.1

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* replace all kustomize versions with 5.2.1

Signed-off-by: Alin Spinu <spinualin@gmail.com>

---------

Signed-off-by: Alin Spinu <spinualin@gmail.com>

Commit:
fix(components): Fix model eval import error in text generation/classification eval pipeline

PiperOrigin-RevId: 621897220

Commit:
chore(components): GCPC 2.13.1 Release

PiperOrigin-RevId: 621998414

dependabot[bot] committed:
chore(deps): bump follow-redirects from 1.6.1 to 1.15.6 in /frontend (#10575)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.6.1 to 1.15.6.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.6.1...v1.15.6)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump pillow from 10.0.1 to 10.3.0 in /backend/src/apiserver/visualization (#10658)

Bumps [pillow](https://github.com/python-pillow/Pillow) from 10.0.1 to 10.3.0.
- [Release notes](https://github.com/python-pillow/Pillow/releases)
- [Changelog](https://github.com/python-pillow/Pillow/blob/main/CHANGES.rst)
- [Commits](https://github.com/python-pillow/Pillow/compare/10.0.1...10.3.0)

---
updated-dependencies:
- dependency-name: pillow
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
fix(components): Remove the unused generate_default_instruction and resolve_upload_location from function_based

PiperOrigin-RevId: 622081254

dependabot[bot] committed:
chore(deps): bump express from 4.18.2 to 4.19.2 in /frontend (#10639)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

chensun committed:
chore: Update argo images to 3.4.16 (#10618)

Signed-off-by: Chen Sun <chensun@users.noreply.github.com>

Commit:
feat(components): Use larger base reward model when tuning `text` and `chat` variants of `bison@001` with the `preview.llm.rlhf_pipeline`

PiperOrigin-RevId: 622229648

dependabot[bot] committed:
chore(deps): bump follow-redirects from 1.5.10 to 1.15.6 in /frontend/server (#10574)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.5.10 to 1.15.6.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.5.10...v1.15.6)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

gregsheremeta committed:
chore(sample): add note about secret needing to be pre-created (#10659)

these samples currently only work on the Google Cloud distribution
because of hardcoded secret name that only pre-exists there.
1 extract the hardcoded secret name to a const to make it a little
easier to change
2 add a note about it.

Signed-off-by: Greg Sheremeta <gshereme@redhat.com>

Tomcli committed:
Chore(components): Clean up old ibm components (#10680)

* remove deprecated ibm components

Signed-off-by: tomcli <tommy.chaoping.li@ibm.com>

* update linkage to the ibm components

Signed-off-by: tomcli <tommy.chaoping.li@ibm.com>

---------

Signed-off-by: tomcli <tommy.chaoping.li@ibm.com>

Commit:
chore(components): Update AutoSxS and RLHF image tags

PiperOrigin-RevId: 623200033

abaland committed:
feat(sdk+backend): Add support for generic ephemeral volume (#10605)

* feat(sdk+backend): Add add_ephemeral_volume method to python sdk + add support to backend
Signed-off-by: abaland <abaland@indeed.com>

* feat(sdk+backend): Add add_ephemeral_volume method to python sdk + add support to backend

Signed-off-by: abaland <abaland@indeed.com>

* chore: upgrade go module + go mod tidy

Signed-off-by: abaland <abaland@indeed.com>

* chore: upgrade license files

Signed-off-by: abaland <abaland@indeed.com>

---------

Signed-off-by: abaland <abaland@indeed.com>

Commit:
feat(components): Use larger base reward model when tuning `t5-xxl` with the `preview.llm.rlhf_pipeline`

PiperOrigin-RevId: 623336791

Tomcli committed:
chore(backend): Promote @rimolive as the backend reviewer (#10689)

Signed-off-by: tomcli <tommy.chaoping.li@ibm.com>

connor-mccarthy committed:
chore(sdk): release `kfp-kubernetes` 1.2.0 (#10692)

Signed-off-by: connor-mccarthy <mccarthy.connor.james@gmail.com>

connor-mccarthy committed:
chore(sdk): make `kfp-kubernetes` release instructions public (#10693)

Signed-off-by: connor-mccarthy <mccarthy.connor.james@gmail.com>

cjidboon94 committed:
chore(kfp-kubernetes): change type of affinity weight to int32 (#10671)

Signed-off-by: Cornelis Boon <cjidboon94@gmail.com>

Commit:
fix(components): Fix image version parameter in rl pipelines

PiperOrigin-RevId: 624348564

Commit:
docs(components): internal

PiperOrigin-RevId: 625114315

Commit:
feat(backend): Merge kfp-tekton backend code (#10678)

* Merge kfp-tekton backend code

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

* Add swf work

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

---------

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

gmfrasca committed:
feat(backend): Upgrade argo to v3.4.16 (#10568)

* pull argo v3.4.16 upstream

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* upgrade to Argo v3.4.16

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* Update ValidateWorkflow calls

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* Add NodeStatus pod name retrieval function

- Argo 3.4.16 upgrade introduces a breaking change with inconsistent node.ID vs
  node.Name
- introduce a function in workflow.go to conditionally handle this

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* Remove PNS Executor manifests and containerRuntimeExecutor ConfigMap Key

- PNS Executor was removed in Argo v3.4, so manifests no longer valid
- WorkflowController will fail to start if `containerRuntimeExecutor`
  provided as input parameter, so remove from WC ConfigMap and CM
  patches

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* fix(frontend): Fix Sidebar tabs to work with argo pod name-id mismatch

- Stemming from upgrade to argo 3.4, Pod Name is no longer always the
  same as NodeID, which breaks a few tabs (PodInfo, PodEvents and
  PodLogs).  Add function to address this

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* test: update frontend CI to accommodate pod id/name changes

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

---------

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

HumairAK committed:
feat(backend): add namespace & prefix scoped credentials to kfp-launcher config for object store paths (#10625)

* add bucket session info to pipeline context

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* allow driver to read bucket session info

Instead of only reading the kfp-launcher when a custom pipeline root is specified, the root dag will now always read the kfp-launcher config to search for a matching bucket if such a configuration is provided in kfp-launcher

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* add support for bucket prefix matching

Provides a structured configuration for bucket providers, whereby user can specify credentials for different providers and path prefixes. A new interface for providing sessions is introduced, which should be implemented for any new provider configuration support.

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* allow object store to handle different providers

Utilizes blob provider specific constructors to open s3, minio, gcs accordingly. If a sessioninfo is provided (via kfp-launcher config) then the associated secret is fetched for each case to gain credentials. If fromEnv is provided, then the standard url opener is used. Also separates out config fields and operations to a separate file.

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* utilize session info in launcher & importer

retrieves the session info (if provided via kfp-launcher) and utilizes it for opening the provider's associated bucket

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* skip config for default aws s3 endpoint

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* chore: refactor/clarify store session info naming

also added some additional code comments clarifying store cred variable usage

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* chore: handle query parameters as s3

as well as update validation logic for provider config, and fix tests
accordingly.

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

---------

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump sqlparse from 0.4.4 to 0.5.0 in /backend/src/apiserver/visualization (#10700)

Bumps [sqlparse](https://github.com/andialbrecht/sqlparse) from 0.4.4 to 0.5.0.
- [Changelog](https://github.com/andialbrecht/sqlparse/blob/master/CHANGELOG)
- [Commits](https://github.com/andialbrecht/sqlparse/compare/0.4.4...0.5.0)

---
updated-dependencies:
- dependency-name: sqlparse
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
feat(components): add resolve_reference_model_metadata to rlhf_preprocessor component

PiperOrigin-RevId: 625496222

TheMichaelHu committed:
feat(components): Move AutoSxS pipeline to v1 directory

PiperOrigin-RevId: 625739392

Commit:
feat(components): add resolve_machine_spec and resolve_refined_image_uri to rlhf_preprocessor component

PiperOrigin-RevId: 626080295

jsondai committed:
fix(components): remove default prediction column names in evaluation classification component to fix incorrect column names for bigquery data source

PiperOrigin-RevId: 626436329

Commit:
feat(components): Introduce placeholders: SERVICE_ACCOUNT_PLACEHOLDER, NETWORK_PLACEHOLDER, PERSISTENT_RESOURCE_ID_PLACEHOLDER and ENCYRPTION_SPEC_KMS_KEY_NAME_PLACEHOLDER. In addition, use PERSISTENT_RESOURCE_ID_PLACEHOLDER as the default value of persistent_resource_id for CustomTrainingJobOp and create_custom_training_job_op_from_component. With this change, custom job created without explicitly setting persistent_resource_id will inherit job level persistent_resource_id, if Persistent Resource is set as job level runtime

PiperOrigin-RevId: 627113501

Commit:
chore: Add kfp-tekton integration tests and manifests (#10702)

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

Commit:
No public description

PiperOrigin-RevId: 627441055

Commit:
chore(components): Drop GCPC Python 3.7

PiperOrigin-RevId: 627500444

Commit:
feat(components): use rlhf_preprocessor to replace the current value_exists call in rlhf

PiperOrigin-RevId: 627630447

Commit:
chore(components): Drop support for Python 3.7 in GCPC

PiperOrigin-RevId: 627783390

Commit:
feat(components): internal

PiperOrigin-RevId: 627804544

Commit:
feat(components): Expand regions supported by `preview.llm.rlhf_pipeline`

PiperOrigin-RevId: 627819341

Commit:
chore(components): Update AutoSxS and RLHF image tags

PiperOrigin-RevId: 627913221

Commit:
chore(components): Change the warning for Python 3.8

PiperOrigin-RevId: 628153803

Commit:
feat(components): migrate function_based resolve_num_microbatches to rlhf_preprocessor component

PiperOrigin-RevId: 628226399

Commit:
chore(components): GCPC 2.14.0 Release

PiperOrigin-RevId: 628241904

Commit:
feat(components): migrate function_based convert_to_delimited_string to rlhf_preprocessor component

PiperOrigin-RevId: 628282787

jsondai committed:
feat(components): Move ModelImportEvaluationOp component to preview namespace

PiperOrigin-RevId: 628459625

Commit:
chore(components): Update AutoSxS and RLHF image tags

PiperOrigin-RevId: 629198194

Commit:
feat(components): migrate function_based resolve_regional_endpoint to rlhf_preprocessor component

PiperOrigin-RevId: 629315370

chensun committed:
chore(backend): update kfp launcher and driver images (#10733)

Signed-off-by: Chen Sun <chensun@users.noreply.github.com>

chensun committed:
chore(release): bumped version to 2.2.0

Commit:
Merge tag '2.2.0' into rebase-2.2.0

Kubeflow Pipelines 2.2.0 release

HumairAK commented:
it seems the pr is missing the code changes introduced by the rebased changes, example: https://github.com/rimolive/data-science-pipelines/blob/rebase-2.2.0/VERSION 
should read 2.2 from this commit https://github.com/opendatahub-io/data-science-pipelines/pull/68/commits/dd59f48cdd0f6cd7fac40306277ef5f3dad6e263#diff-7b60b8e351cbb80c47459ffe2c79f1a26404871f49294780fe47ad0e58c09350
